tools/pybug.py

```diff
@@ -0,0 +1,1544 @@
+# Copyright (c) 2013-2020 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+""" An application for booting SpiNNaker chips
+"""
+from __future__ import print_function  # for 2.7
+from collections import namedtuple
+from datetime import datetime
+from enum import IntEnum
+import re
+import struct
+import sys
+import time
+from zlib import crc32
+from spinn_front_end_common import __version__ as fec_version
+from tools.exn import BadArgs, SpinnException, SpinnTooManyRetriesException
+from tools.cli import CLI
+from tools.boot import boot
+from tools.sv import Struct
+from tools.cmd import SCAMPCmd, BMPCmd, SCAMP_CMD
+from tools.util import (
+    read_file, hex_dump, parse_cores, parse_region, parse_apps, parse_bits)
+
+# This code ought to be rewritten to use the Cmd package:
+# https://docs.python.org/3.8/library/cmd.html
```

Comment:
Why not do it now? 

And if the answer is because it is messy with python2 then I vote that this new feature is not supported in python 2

Reply:
I think I'd describe that as a longer-term plan. It's quite a large change as it works in a different way.

I found that package when looking up how to make readline (itself not one of the better documented Python packages) work.

Reply:
Since this is a disguised call for doing a lot of work, I _reject_ your comment as relevant for this PR.